input/system/selfhosted/otel_handler.go

```diff
@@ -115,9 +120,18 @@ func setupOtelHandler(ctx context.Context, server *state.Server, rawLogStream ch
 				prefixedLogger.PrintError("Could not read otel body")
 			}
 			logsData := &otlpLogs.LogsData{}
-			if err := proto.Unmarshal(b, logsData); err != nil {
-				prefixedLogger.PrintError("Could not unmarshal otel body")
+			contentType := r.Header.Get("Content-Type")
+			switch contentType {
+			case "application/json":
+				if err := protojson.Unmarshal(b, logsData); err != nil {
+					prefixedLogger.PrintError("Could not unmarshal otel body, json expected")
```

Comment:
Couldn't we add here the `err.Error()` for clarity. In my experience it was hard to understand what was the issue without running locally the patched collector. 

Of course, protobuf's error `wire protocol error` wouldn't help much though.

Reply:
Agreed that its helpful to log more here - in the rebased version (https://github.com/pganalyze/collector/pull/747) I log both the error message and the original data that we received that we failed to unmarshal.